### PR TITLE
Make the node binaries available using cabal

### PR DESCRIPTION
**Pull request by raduom**
Make the node and cli binaries available using cabal.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by raduom** [comment]
Hold on to merging this. I still want to make a clean build to make sure all dependencies are built properly (libsodium, i am looking at you!).

File: plutus-pab/plutus-pab.cabal [hunk 1]
```diff
@@ -39,6 +39,7 @@ common lang
 
 library
     import: lang
+    build-tool-depends: cardano-node:cardano-node, cardano-cli:cardano-cli
```
Comment (michaelpj) [inline plutus-pab/plutus-pab.cabal:42]: Shouldn't this be on the _tests_? I'm not sure that putting it as a build dependency on the library will do what you want. If you have it as a build tool dependency of tests, I think that you should have them in scope and you shouldn't need the `cabal exec` in the scripts.

Comment (raduom) [reply plutus-pab/plutus-pab.cabal:42]: I moved it to `plutus-pab-test-light`, but I still need to call it using `cabal exec`.

Comment (raduom) [reply plutus-pab/plutus-pab.cabal:42]: Right now I just want to start everything manually. Have access to those binaries in a nix-shell.

Comment (michaelpj) [reply plutus-pab/plutus-pab.cabal:42]: Ah I see. Yeah, using `cabal exec` for that is a nice solution.

Comment (michaelpj) [reply plutus-pab/plutus-pab.cabal:42]: Then you maybe don't even need this dependency, but it can't hurt.

Comment (raduom) [reply plutus-pab/plutus-pab.cabal:42]: Tried without and it cannot `cabal exec cardano-cli`.

**Pull request merged**